pkg/workspace/workspace_set.go

```diff
@@ -6,46 +6,32 @@ import (
 	"github.com/warpfork/warpforge/wfapi"
 )
 
-// a workspace set consists of the 3 types of workspace we operate on
-//  home: a workspace containing configuration information and other global info
-//  root: a workspace containing catalogs to use, which also stores wares and cache
-//        the home workspace is the default root workspace
-//  stack: a set of workspaces that may contain additional catalogs and other project-specific info
-type WorkspaceSet struct {
-	Home  *Workspace
-	Root  *Workspace
-	Stack []*Workspace
+// A WorkspaceSet is a slice of workspaces ending with a root workspace.
+type WorkspaceSet []*Workspace
+
+func (wsSet WorkspaceSet) Root() *Workspace {
+	return wsSet[len(wsSet)-1]
 }
 
 // Opens a full WorkspaceSet
 // searches from searchPath up to basisPath for workspaces
-// root workspace will be the first workspace found that is marked as a root, or the home workspace if none exists
+// The root workspace will be the first workspace found that is marked as a root, or the home workspace if none exists
 // Errors:
 //
 //    - warpforge-error-workspace -- when the workspace directory fails to open
 //    - warpforge-error-searching-filesystem -- when an error occurs while searching for the workspace
 func OpenWorkspaceSet(fsys fs.FS, basisPath string, searchPath string) (WorkspaceSet, wfapi.Error) {
```

Comment:
Can we just nuke this function entirely now, then?  FindWorkspaceStack seems to already do everything.

In particular, FindWorkspaceStack is already doing the "include home workspace if no other root found" check.

Reply:
I think I lean toward removing the `FindWorkspace*` functions all together in favor of dumping that into the `OpenWorkspace*` functions. The `Find` functions are calling `openWorkspace` anyway.

Reply:
If it's just naming at that point... I think the `Find*` names make more sense than the `Open*` names, with what is being done in these functions.  There's not a lot of opening going on; there's no closing to be done afterwards either.

I'd tend to expect a function named `Open*` to be opening files (not just stat'ing their existence), and probably outright hanging on to open handles afterwards.

Reply:
Perhaps the other way then. But `Open` still feels more correct than "Find" for the base `OpenWorkspace`, `OpenHomeWorkspace`, and `OpenRootWorkspace`. Perhaps `Get` would be fine. `New` feels incorrect because it fails when it would have to do a `Create` operation.

Reply:
Still think `Find` describes what's currently happening, which is: it returns a list of paths that look like workspaces.

When we do have workspace config that has some richness: then we'll end up having an `OpenWorkspace` operation that opens some files, reads and parses them, and perhaps does other additional thinking.  But that will be a distinct operation from finding the workspaces: we probably wouldn't even run it on every path returned by `FindWorkspace` in the course of many commands, I suspect.

Reply:
It seems like there's a dichotomy between what happens internally "create a new workspace struct" and externally "there should be a file here and it is an error if there isn't"